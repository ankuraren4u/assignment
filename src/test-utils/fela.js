import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createRenderer } from "fela";
import { RendererProvider, ThemeProvider } from "react-fela";

import { theme } from "./../theme";
Enzyme.configure({ adapter: new Adapter() });

const renderer = createRenderer();

export const felaShallow = node =>
  shallow(
    <RendererProvider renderer={renderer}>
      <ThemeProvider theme={theme}>{node}</ThemeProvider>
    </RendererProvider>
  );

export const felaMount = node =>
  mount(
    <RendererProvider renderer={renderer}>
      <ThemeProvider theme={theme}>{node}</ThemeProvider>
    </RendererProvider>
  );
