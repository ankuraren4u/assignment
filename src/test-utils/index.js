export { felaShallow, felaMount } from "./fela";
