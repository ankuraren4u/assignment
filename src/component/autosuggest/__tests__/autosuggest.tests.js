import { Autosuggest } from "..";
import { felaShallow } from "./../../../test-utils/";

describe("<Autosuggest />", () => {
  it("Init Test", () => {
    const suggestions = [];
    const onQueryChange = jest.fn();
    const onSelectionChange = jest.fn();

    const component = felaShallow(
      <Autosuggest
        onQueryChange={onQueryChange}
        suggestions={suggestions}
        onSelectionChange={onSelectionChange}
      />
    );
    expect(component.html()).toMatchSnapshot();
    /** TODO : Write tests */
  });
});
