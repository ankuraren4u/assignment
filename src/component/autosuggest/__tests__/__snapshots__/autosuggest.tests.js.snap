// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Autosuggest /> Init Test 1`] = `"<div class=\\"ui-autosuggest a b c d e f\\"><input type=\\"text\\" value=\\"\\" class=\\"g h i b j k l\\"/></div>"`;
